packages/ember/tests/routing/basic_test.js

```diff
@@ -1931,8 +1932,24 @@ QUnit.test('Route should tear down multiple outlets', function() {
   equal(jQuery('div.posts-footer:contains(postsFooter)', '#qunit-fixture').length, 0, 'The posts/footer template was removed');
 });
 
+QUnit.test('Route will assert if you try to explicitly render {into: ...} a {{render}} helper that resolves to an {{outlet}}', function() {
+
+  Router.map(function() {
+    this.route('home', { path: '/' });
+  });
+
+  App.HomeRoute = Route.extend({
+    renderTemplate() {
+      this.render({ into: 'nonexistent' });
+    }
+  });
+
+  expectAssertion(() => bootApplication(), 'Rendering into a {{render}} helper that resolves to an {{outlet}} is deprecated.');
+});
+
 
 QUnit.test('Route will assert if you try to explicitly render {into: ...} a missing template', function () {
+  ENV._ENABLE_ORPHANED_OUTLETS_SUPPORT = true;
```

Comment:
Can we reset this value in the teardown like this (https://github.com/emberjs/ember.js/pull/15900/files#diff-7c1b00291152c89da72c8940b7305c1a)

Reply:
Good call! Sorry I missed that 👍